### PR TITLE
Add author identification and warning mechanism

### PR DESCRIPTION
**Pull request by gagyibenedek**

### CODE REVIEW

**Comment by gagyibenedek** [comment]
Fixed the merge conflicts in the GH online editor, which I haven't used until now, so I hope I haven't broken anything :)

File: lib/marshalls/author.marshall.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  title () {
+    return 'Identifying package author...'
+  }
+
+  validate (pkg) {
+    return this.packageRepoUtils.getPackageInfo(pkg.packageName).then(data => {
+      let lastVersion
+      for (let i in data.versions) {
```
[the remark this answers is not in the document]
Comment (gagyibenedek) [reply lib/marshalls/author.marshall.js]: it's not an array :( it's just an object with a bunch of properties.

[the remark this answers is not in the document]
Comment (lirantal) [reply lib/marshalls/author.marshall.js]: ahh my bad
how about you find the latest version using the `dist-tags['latest']` and then access the relevant version in versions?

check this for example: http://registry.npmjs.org/npq

[the remark this answers is not in the document]
Comment (gagyibenedek) [reply lib/marshalls/author.marshall.js]: that sounds much better :) will fix it tomorrow

**Comment by gagyibenedek** [comment]
The 'maintainers' field is non-optional,  it's filled automatically with the data the npm user is registered. So checking that has no information value from our perspective.
The author field on the other hand is not mandatory, so it missing may be a sign of malicious intent. Since some  popular packages lack it too, it's not an error, just a warning.  It's like saying: 'please double-check this, but if you're sure, you can proceed.'

**Comment by lirantal** [comment]
Ahh, good point about the maintainer field being set up automatically for you.

**Pull request merged**